File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        sub = set()
        j, l, m = 0, 0, 0

        while j < len(s):
            while s[j] in sub:
                sub.remove(s[l])
                l += 1
            sub.add(s[j])
            m = max(m, j - l + 1)
            j += 1
        return m

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
const lengthOfLongestSubstring = (s) => {
  const n = s.length
  let maxLen = 0
  let maxStr = ''

  for (let i = 0; i < n; i++) {
    let temp = s[i]
    let posIndex = maxStr.indexOf(temp)
    
    if (posIndex > -1) {
      maxStr = maxStr.substring(posIndex + 1)
    }

    maxStr += temp
    maxLen = Math.max(maxLen, maxStr.length)
  }
  return maxLen
}